src/async_impl/client.rs

```diff
@@ -1140,6 +1148,28 @@ impl ClientBuilder {
         self
     }
 
+    /// Controls the use of server name indication.
+    ///
+    /// Defaults to `true`.
+    ///
+    /// # Optional
+    ///
+    /// This requires the optional `default-tls`, `native-tls`, or `rustls-tls(-...)`
+    /// feature to be enabled.
+    #[cfg(feature = "__tls")]
+    #[cfg_attr(
+        docsrs,
+        doc(cfg(any(
+            feature = "default-tls",
+            feature = "native-tls",
+            feature = "rustls-tls"
+        )))
+    )]
+    pub fn use_sni(mut self, use_sni: bool) -> ClientBuilder {
```

Comment:
My initial reaction was to name this method `tls_sni`. I can't claim it's objectively better, maybe use sni feels more natural. But want to just toss it out there.

Reply:
Yes I think it is more clear to prepend with tls as SNI may not be known to the reader.